// @awesome-cordova-plugins/barcode-scanner/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { Injectable } from '@angular/core';
import { AwesomeCordovaNativePlugin, cordova } from '@awesome-cordova-plugins/core';
import * as ɵngcc0 from '@angular/core';
var BarcodeScanner = /** @class */ (function (_super) {
    __extends(BarcodeScanner, _super);
    function BarcodeScanner() {
        var _this = _super !== null && _super.apply(this, arguments) || this;
        _this.Encode = {
            TEXT_TYPE: 'TEXT_TYPE',
            EMAIL_TYPE: 'EMAIL_TYPE',
            PHONE_TYPE: 'PHONE_TYPE',
            SMS_TYPE: 'SMS_TYPE',
        };
        return _this;
    }
    BarcodeScanner.prototype.scan = function (options) { return cordova(this, "scan", { "callbackOrder": "reverse" }, arguments); };
    BarcodeScanner.prototype.encode = function (type, data) { return cordova(this, "encode", {}, arguments); };
    BarcodeScanner.pluginName = "BarcodeScanner";
    BarcodeScanner.plugin = "phonegap-plugin-barcodescanner";
    BarcodeScanner.pluginRef = "cordova.plugins.barcodeScanner";
    BarcodeScanner.repo = "https://github.com/phonegap/phonegap-plugin-barcodescanner";
    BarcodeScanner.platforms = ["Android", "BlackBerry 10", "Browser", "iOS", "Windows"];
BarcodeScanner.ɵfac = /*@__PURE__*/ function () { var ɵBarcodeScanner_BaseFactory; return function BarcodeScanner_Factory(t) { return (ɵBarcodeScanner_BaseFactory || (ɵBarcodeScanner_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(BarcodeScanner)))(t || BarcodeScanner); }; }();
BarcodeScanner.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: BarcodeScanner, factory: function (t) { return BarcodeScanner.ɵfac(t); } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BarcodeScanner, [{
        type: Injectable
    }], null, null); })();
    return BarcodeScanner;
}(AwesomeCordovaNativePlugin));
export { BarcodeScanner };
