// @ionic/angular-server/bundles/ionic-angular-server.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/common'), require('@angular/core'), require('@angular/platform-server'), require('@ionic/core/hydrate')) :
    typeof define === 'function' && define.amd ? define('@ionic/angular-server', ['exports', '@angular/common', '@angular/core', '@angular/platform-server', '@ionic/core/hydrate'], factory) :
    (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory((global.ionic = global.ionic || {}, global.ionic["angular-server"] = {}), global.ng.common, global.ng.core, global.ng.platformServer, global.hydrate));
})(this, (function (exports, common, i0, platformServer, hydrate) { 'use strict';

    function _interopNamespace(e) {
        if (e && e.__esModule) return e;
        var n = Object.create(null);
        if (e) {
            Object.keys(e).forEach(function (k) {
                if (k !== 'default') {
                    var d = Object.getOwnPropertyDescriptor(e, k);
                    Object.defineProperty(n, k, d.get ? d : {
                        enumerable: true,
                        get: function () { return e[k]; }
                    });
                }
            });
        }
        n["default"] = e;
        return Object.freeze(n);
    }

    var i0__namespace = /*#__PURE__*/_interopNamespace(i0);

    // @dynamic
    var IonicServerModule = /** @class */ (function () {
        function IonicServerModule() {
        }
        return IonicServerModule;
    }());
    /** @nocollapse */ IonicServerModule.ɵfac = i0__namespace.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: IonicServerModule, deps: [], target: i0__namespace.ɵɵFactoryTarget.NgModule });
    /** @nocollapse */ IonicServerModule.ɵmod = i0__namespace.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: IonicServerModule });
    /** @nocollapse */ IonicServerModule.ɵinj = i0__namespace.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: IonicServerModule, providers: [
            {
                provide: platformServer.BEFORE_APP_SERIALIZED,
                useFactory: hydrateIonicComponents,
                multi: true,
                deps: [common.DOCUMENT, i0.APP_ID],
            },
        ] });
    i0__namespace.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.14", ngImport: i0__namespace, type: IonicServerModule, decorators: [{
                type: i0.NgModule,
                args: [{
                        providers: [
                            {
                                provide: platformServer.BEFORE_APP_SERIALIZED,
                                useFactory: hydrateIonicComponents,
                                multi: true,
                                deps: [common.DOCUMENT, i0.APP_ID],
                            },
                        ],
                    }]
            }] });
    // @dynamic
    function hydrateIonicComponents(doc, appId) {
        // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
        return function () {
            return hydrate.hydrateDocument(doc, {
                clientHydrateAnnotations: false,
                excludeComponents: [
                    // overlays
                    'ion-action-sheet',
                    'ion-alert',
                    'ion-loading',
                    'ion-modal',
                    'ion-picker',
                    'ion-popover',
                    'ion-toast',
                    'ion-toast',
                    // navigation
                    'ion-router',
                    'ion-route',
                    'ion-route-redirect',
                    'ion-router-link',
                    'ion-router-outlet',
                    // tabs
                    'ion-tabs',
                    'ion-tab',
                    // auxiliar
                    'ion-picker-column',
                    'ion-virtual-scroll',
                ],
            }).then(function (hydrateResults) {
                hydrateResults.diagnostics.forEach(function (d) {
                    if (d.type === 'error') {
                        console.error(d.messageText);
                    }
                    else if (d.type === 'debug') {
                        console.debug(d.messageText);
                    }
                    else {
                        console.log(d.messageText);
                    }
                });
                if (doc.head != null) {
                    var styleElms = doc.head.querySelectorAll('style[data-styles]');
                    // eslint-disable-next-line @typescript-eslint/prefer-for-of
                    for (var i = 0; i < styleElms.length; i++) {
                        styleElms[i].setAttribute('ng-transition', appId);
                    }
                }
                if (doc.body != null) {
                    var ionPages = doc.body.querySelectorAll('.ion-page.ion-page-invisible');
                    // eslint-disable-next-line @typescript-eslint/prefer-for-of
                    for (var i = 0; i < ionPages.length; i++) {
                        ionPages[i].classList.remove('ion-page-invisible');
                    }
                }
            });
        };
    }

    /**
     * Generated bundle index. Do not edit.
     */

    exports.IonicServerModule = IonicServerModule;

    Object.defineProperty(exports, '__esModule', { value: true });

}));
